engine/consensus/matching/engine.go

```diff
@@ -654,17 +640,52 @@ func (e *Engine) sealResult(result *flow.ExecutionResult, final *flow.Header) er
 	return nil
 }
 
-func (e *Engine) clearPools(sealedIDs []flow.Identifier) {
+// collectAggregateSignatures collects approver signatures and identities per
+// chunk
+// TODO: needs testing
+func (e *Engine) collectAggregateSignatures(result *flow.ExecutionResult) []flow.AggregatedSignature {
+	resultID := result.ID()
+	signatures := make([]flow.AggregatedSignature, 0, len(result.Chunks))
+
+	for _, chunk := range result.Chunks {
+		// get approvals for result with chunk index
+		approvals := e.approvals.ByChunk(resultID, chunk.Index)
+		if approvals == nil {
+			continue
+		}
+
+		// temp slices to store signatures and ids
+		sigs := make([]crypto.Signature, 0, len(approvals))
+		ids := make([]flow.Identifier, 0, len(approvals))
+
+		// for each approval collect signature and approver id
+		for _, approval := range approvals {
+			ids = append(ids, approval.Body.ApproverID)
+			sigs = append(sigs, approval.Body.AttestationSignature)
+		}
+
+		aggSign := flow.AggregatedSignature{
+			VerifierSignatures: sigs,
+			SignerIDs:          ids,
+		}
+
+		signatures = append(signatures, aggSign)
+	}
 
-	// NOTE: Below, we clear our memory pool of all entities that reference
-	// blocks at or below the sealed head. If we don't know the block, we purge
-	// the entities once we have sealed a further 100 blocks without seeing the
-	// block (it's probably no longer a valid extension of the state anyway).
+	return signatures
+}
+
+// clearPools clears the memory pools of all entities related to blocks that are
+// already sealed. If we don't know the block, we purge the entities once we
+// have sealed a further 100 blocks without seeing the block (it's probably no
```

Comment:
⚠️ 
I am worried that the implementation does not conform to the documentation
> we purge the entities once we have sealed a further 100 blocks without seeing the block

Reasoning:
- we increase the counter during each execution of this method: https://github.com/onflow/flow-go/blob/4b7c696b01e7881db12631f7a49d6ed4123c1e54/engine/consensus/matching/engine.go#L745-L747
- Couldn't this method be called for each received result Approval? Therefore, 100 executions might be performed for a single block if we have 100 or more verifiers.